pkg/operator/osd/storage.go

```diff
@@ -94,6 +94,9 @@ func (s *StorageSpec) resolveNodeConfig(node *Node) {
 	resolveInt(&(node.Config.StoreConfig.WalSizeMB), s.Config.StoreConfig.WalSizeMB, 0)
 	resolveInt(&(node.Config.StoreConfig.JournalSizeMB), s.Config.StoreConfig.JournalSizeMB, 0)
 	resolveString(&(node.Config.Location), s.Config.Location, "")
+	if len(node.Directories) == 0 {
```

Comment:
to be more accurate, this should be in the `resolveNodeSelection` function instead.